src/components/Post/PostScreen.js

```diff
@@ -0,0 +1,155 @@
+import React, { useState } from 'react';
+import {
+  Text, Button, View, StyleSheet,
+} from 'react-native';
+import PropTypes from 'prop-types';
+import { ScrollView } from 'react-native-gesture-handler';
+import firestore from '@react-native-firebase/firestore';
+import Post from './Post';
+import Comment from './Comment';
+
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+  },
+  welcomeMessage: {
+    textAlign: 'center',
+    textAlignVertical: 'center',
+    height: 50,
+    fontSize: 20,
+  },
+});
+
+function CommentLoader({ postID }) {
+  const [errorMessage, setErrorMessage] = useState(null);
+  const [commentList, setCommentList] = useState([]);
+  let commentsData = [];
+
+  firestore().collection('comments')
```

Comment:
we can get all the comments in a collection more efficiently by performing a composite index query:

`Firestore().collection('comments').where('postId', '==', postId).orderBy('createdAt', 'desc')`

the above code gets all comments where the postId is equal to the parameter you pass into commentLoader, and orders all results by the timestamp, in descending order.

Right now, your code gets all the comments in the posts collection, which is not scalable long-term. We can take advantage of Firestore's powerful query results this way, instead of manually filtering and sorting the results ourselves.

We will later explore paginating the data so we do not return all the comments for every post, and paginate the results by (5 comments-ish) or so, so we do not quickly exceed the amount of document reads we have for Firestore

Reply:
the above code will replace lines 29-52 btw